app/js/invertedIndex.js

```diff
@@ -1,67 +1,106 @@
+/**
+ * InvertedIndex class
+ * @class
+ */
 class InvertedIndex {
-
-  constructor(){
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
     this.indexed = {};
   }
 
+  /**
+   * Get the index for a particular file
+   * @function
+   * @param {String} fileName
+   * @return {Object} index object
+   */
   getIndex(fileName) {
-      return this.indexed[fileName];
+    return this.indexed[fileName];
   }
 
-  readFile(file) {
+  /**
+   * Read files using FileReader
+   * @function
+   * @param {String} file
+   * @return {Promise} validateFile response
+   */
+  static readFile(file) {
     return new Promise((resolve, reject) => {
```

Comment:
Thank you. Actually, it can also be implemented from the front end. But the requirement was to have the readFile as part of the class. Also, I wanted to ensure the file is validated before any other thing, hence the use of promise.